R/compareScenarios.R

```diff
@@ -1603,33 +1601,79 @@ compareScenarios <- function(mif, hist,
   targets = c("Emi|GHGtot|target|40% (Mt CO2-equiv/yr)"="-40% by 2030 (vs. 1990)",
               "Emi|GHGtot|target|55% (Mt CO2-equiv/yr)"="-55% by 2030 (vs. 1990)",
               "Emi|GHGtot|target|65% (Mt CO2-equiv/yr)"="-65% by 2030 (vs. 1990)")
+  
+  # remove as soon as renaming is done
+  if ("Emi|GHGtot (Mt CO2-equiv/yr)" %in% getNames(histData, dim=3)) {
+    ghg_var <- "Emi|GHGtot (Mt CO2-equiv/yr)"  # old name
+  } else {
+    ghg_var <- "Emi|GHG (Mt CO2eq/yr)"  # new name
+  }
 
-  p <- mipLineHistorical(data[mainReg,,"Emi|GHG (Mt CO2eq/yr)"],x_hist=histData[mainReg,,"Emi|GHGtot (Mt CO2-equiv/yr)"],
+  p <- mipLineHistorical(data[mainReg,,"Emi|GHG (Mt CO2eq/yr)"],x_hist=histData[mainReg,,ghg_var],
                          ylab='Total GHG Emisions w/o Bunkers [Mt CO2-equiv/yr]',scales="free_y",plot.priority=c("x_hist","x","x_proj"),
                          hlines=if(all(names(targets) %in% getNames(histData, dim=3)) && (!(any(is.na(histData[mainReg,2030,names(targets)]))))) histData[mainReg,2030,names(targets)] else NULL,
                          hlines.labels=targets)
   swfigure(sw,print,p,sw_option="height=8,width=8")  
-  
-  p <- mipLineHistorical(data[,,"Emi|GHG (Mt CO2eq/yr)"][mainReg,,,invert=TRUE],x_hist=histData[,,"Emi|GHGtot (Mt CO2-equiv/yr)"][mainReg,,,invert=TRUE],
+
+  p <- mipLineHistorical(data[,,"Emi|GHG (Mt CO2eq/yr)"][mainReg,,,invert=TRUE],x_hist=mbind(histData[,,ghg_var][mainReg,,,invert=TRUE],histData[,,ghg_var][mainReg,,,invert=TRUE]),
                          ylab='Total GHG Emisions w/o Bunkers [Mt CO2-equiv/yr]',scales="free_y",plot.priority=c("x_hist","x","x_proj"),
                          hlines=if(all(names(targets) %in% getNames(histData, dim=3)) && (!(any(is.na(histData[,2030,names(targets)][mainReg,,,invert=TRUE]))))) histData[,2030,names(targets)][mainReg,,,invert=TRUE] else NULL,
                          hlines.labels=targets)
   swfigure(sw,print,p,sw_option="height=9,width=8")
-  
-  
+
+
   #GHG total with bunkers
   p <- mipLineHistorical(data[mainReg,,"Emi|GHG|w/ Bunkers (Mt CO2eq/yr)"],x_hist=NULL,
                          ylab='Total GHG Emisions w/ Bunkers [Mt CO2-equiv/yr]',scales="free_y",plot.priority=c("x_hist","x","x_proj"))
   swfigure(sw,print,p,sw_option="height=8,width=8")
-  
+
   p <- mipLineHistorical(data[,,"Emi|GHG|w/ Bunkers (Mt CO2eq/yr)"][mainReg,,,invert=TRUE],x_hist=NULL,
                          ylab='Total GHG Emisions w/ Bunkers [Mt CO2-equiv/yr]',scales="free_y",plot.priority=c("x_hist","x","x_proj"),facet.ncol=3)
   swfigure(sw,print,p,sw_option="height=9,width=8")
 
+  ## ---- Emissions GHG Sectors ----
+  items <- c("Emi|GHG|Energy (Mt CO2eq/yr)",
```

Comment:
Would be good to plot Emi|GHG|Land-Use Change here as well although there is no historic data for this yet to have all top-level IPCC sectors in this section.

Reply:
on it